### PR TITLE
Use const sym where possible

### PR DESCRIPTION
**Pull request by camsteffen**
I ran a regex search and replace to use const `sym` values where possible. This should give some performance boost by avoiding string interning at runtime.

Con: It is not as consistent as always using `sym!`.

I also changed an internal lint to suggest using `sym::{}`, making an assumption that this will always work for diagnostic items.

changelog: none

### CODE REVIEW

File: clippy_lints/src/matches.rs [hunk 1]
```diff
@@ -1509,6 +1509,7 @@ mod redundant_pattern_match {
     use rustc_errors::Applicability;
     use rustc_hir::{Arm, Expr, ExprKind, MatchSource, PatKind, QPath};
     use rustc_lint::LateContext;
+    use rustc_span::sym;
```
Comment (flip1995) [inline clippy_lints/src/matches.rs:1512]:
```suggestion
```

Shouldn't be necessary, since it is already imported above.

**Review by flip1995 — COMMENTED** [review]
Thanks! One NIT left.

If you're interested in pushing this even further, we could make an internal lint out of this, which searches for `sym!(xyz)` and checks if a constant named `xyz` exists in `sym::*`. (Definitely in another PR though)

> I also changed an internal lint to suggest using `sym::{}`, making an assumption that this will always work for diagnostic items.

This should be fine.

File: clippy_lints/src/matches.rs (hunk 1 above)
Comment (camsteffen) [reply clippy_lints/src/matches.rs:1512]: It's needed for the inner mod.

**Comment by camsteffen** [comment]
> If you're interested in pushing this even further, we could make an internal lint out of this

I actually started to look into that. I haven't been able to find a way to query for all the available `sym::*` items. There must be a way to get that from the hir?

**Comment by flip1995** [comment]
@bors r+

Thanks!



>  There must be a way to get that from the hir?

Thinking about this more, I can't find a way how this could be done. I don't think there is a query, where you can just throw in any path (like `rustc_span::symbol::sym`) and get the resolved module/item. So this is a bigger challenge, than I initially thought.

**Pull request merged**